odxtools/diaglayercontainer.py

```diff
@@ -124,17 +124,18 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
         for ecu_variant in self.ecu_variants:
             ecu_variant._resolve_odxlinks(odxlinks)
 
-    def _finalize_init(self, odxlinks: OdxLinkDatabase) -> None:
+    def _finalize_init(self, odxlinks: OdxLinkDatabase,
+            ecu_shared_datas: NamedItemList[DiagLayer]) -> None:
         for ecu_shared_data in self.ecu_shared_datas:
-            ecu_shared_data._finalize_init(odxlinks)
+            ecu_shared_data._finalize_init(odxlinks, ecu_shared_datas)
```

Comment:
as mentioned above, it is better pass the whole `Database` object here (if this is *really* necessary.)